packages/authorization-server/test/api/authorizationServer.api.test.ts

```diff
@@ -0,0 +1,154 @@
+import request from "supertest";
+import { describe, it, expect, vi, beforeEach } from "vitest";
+import {
+  ClientId,
+  generateId,
+  makeTokenGenerationStatesClientKidPK,
+  PurposeId,
+} from "pagopa-interop-models";
+import { authorizationServerApi } from "pagopa-interop-api-clients";
+import {
+  InteropConsumerToken,
+  InteropJwtConsumerPayload,
+  InteropJwtHeader,
+} from "pagopa-interop-commons";
+import { getMockClient } from "pagopa-interop-commons-test/index.js";
+import { api, tokenService } from "../vitest.api.setup.js";
+import {
+  clientAssertionRequestValidationFailed,
+  clientAssertionSignatureValidationFailed,
+  clientAssertionValidationFailed,
+  platformStateValidationFailed,
+  tokenGenerationStatesEntryNotFound,
+} from "../../src/model/domain/errors.js";
+
+describe("POST /authorization-server/token.oauth2", () => {
+  const clientId = getMockClient().id;
+  const tokenClientKidPK = makeTokenGenerationStatesClientKidPK({
+    clientId,
+    kid: "",
+  });
+
+  const header: InteropJwtHeader = {
+    alg: "RS256",
+    use: "sig",
+    typ: "at+jwt",
+    kid: generateId(),
+  };
+
+  const payload: InteropJwtConsumerPayload = {
+    jti: generateId(),
+    iss: "interop.pagopa.it",
+    aud: ["interop.pagopa.it"],
+    iat: 10,
+    nbf: 100,
+    exp: 100,
+    client_id: clientId,
+    sub: generateId<ClientId>(),
+    purposeId: generateId<PurposeId>(),
+  };
+
+  const token: InteropConsumerToken = {
+    header,
+    payload,
+    serialized: "",
+  };
+
+  const validRequestBody: authorizationServerApi.AccessTokenRequest = {
+    client_id: "e58035ce-c753-4f72-b613-46f8a17b71cc",
+    client_assertion: "valid-jws-token",
+    client_assertion_type:
+      "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
+    grant_type: "client_credentials",
+  };
+
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+  const makeRequest = (
+    body: authorizationServerApi.AccessTokenRequest = validRequestBody
+  ) =>
+    request(api)
+      .post("/authorization-server/token.oauth2")
+      .set("Content-Type", "application/x-www-form-urlencoded")
+      .send(new URLSearchParams(body as Record<string, string>).toString());
```

Comment:
```suggestion
      .send(body);
```
passing `body` should be enough